### ISSUE TITLE
Can't start mysql 5.6 after working on 5.7

### ISSUE THREAD

**Issue by masiorama** [cytopia/devilbox#778]
### ISSUE TYPE

 - Bug Report

### Checklist

* [x] `.env` file is attached [env.txt](https://github.com/cytopia/devilbox/files/5770914/env.txt)
* [ ] `./check-config.sh` not available on my devilbox version 1.6.2
* [ ] `docker-compose logs` N/A
* [ ] `docker-compose.override.yml` is attached (if exists) N/A
* [ ] Custom configs from `cfg/` dir are attached (if exist) N/A
* [x] I've looked through the docs: https://devilbox.readthedocs.io/en/latest/
* [x] I've looked through existing issues: https://github.com/cytopia/devilbox/issues
* [x] I've read troubleshooting: https://devilbox.readthedocs.io/en/latest/support/troubleshooting.html

### OS / ENVIRONMENT

1. Host operating system and version: Windows 10 Pro
2. (Windows only) Native Docker or Docker Toolbox: Native Docker
3. Docker version: 2.3.0.3
4. Docker Compose version: 1.25.5
5. (Linux) Is SELinux enabled?:
6. What git commit hash are you on?:
7. devilbox version 1.6.2

### SUMMARY

I worked without issues for months with mysql 5.6. I switched for a single project to mysql 5.7. Now I can start 5.7 but cannot start again 5.6 (even if I stop, kill images and restart docker)... it seems that 5.6 and 5.7 are somehow merged (and the mysql5.7 log directory is missing).
The log on starting is this:

```
Attaching to devilbox_mysql_1
mysql_1  | 2021-01-05 16:46:46+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.6.46-1debian9 started.
mysql_1  | 2021-01-05 16:46:46+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
mysql_1  | 2021-01-05 16:46:46+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.6.46-1debian9 started.
mysql_1  | 2021-01-05 16:46:46 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
mysql_1  | 2021-01-05 16:46:46 0 [Note] mysqld (mysqld 5.6.46) starting as process 1 ...
mysql_1  | 2021-01-05 16:46:46 1 [Note] Plugin 'FEDERATED' is disabled.
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Using atomics to ref count buffer pool pages
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: The InnoDB memory heap is disabled
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Memory barrier is not used
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Compressed tables use zlib 1.2.11
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Using Linux native AIO
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Using CPU crc32 instructions
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Initializing buffer pool, size = 128.0M
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Completed initialization of buffer pool
mysql_1  | 2021-01-05 16:46:46 1 [Note] InnoDB: Highest supported file format is Barracuda.
mysql_1  | InnoDB: No valid checkpoint found.
mysql_1  | InnoDB: If you are attempting downgrade from MySQL 5.7.9 or later,
mysql_1  | InnoDB: please refer to http://dev.mysql.com/doc/refman/5.6/en/upgrading-downgrading.html
mysql_1  | InnoDB: If this error appears when you are creating an InnoDB database,
mysql_1  | InnoDB: the problem may be that during an earlier attempt you managed
mysql_1  | InnoDB: to create the InnoDB data files, but log file creation failed.
mysql_1  | InnoDB: If that is the case, please refer to
mysql_1  | InnoDB: http://dev.mysql.com/doc/refman/5.6/en/error-creating-innodb.html
mysql_1  | 2021-01-05 16:46:46 1 [ERROR] Plugin 'InnoDB' init function returned error.
mysql_1  | 2021-01-05 16:46:46 1 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
mysql_1  | 2021-01-05 16:46:46 1 [ERROR] Unknown/unsupported storage engine: InnoDB
mysql_1  | 2021-01-05 16:46:46 1 [ERROR] Aborting
mysql_1  |
mysql_1  | 2021-01-05 16:46:46 1 [Note] Binlog end
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'partition'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'PERFORMANCE_SCHEMA'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_DATAFILES'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_TABLESPACES'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN_COLS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_FIELDS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_COLUMNS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_INDEXES'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_TABLESTATS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_SYS_TABLES'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_INDEX_TABLE'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_INDEX_CACHE'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_CONFIG'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_BEING_DELETED'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_DELETED'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_FT_DEFAULT_STOPWORD'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_METRICS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_BUFFER_POOL_STATS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE_LRU'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX_RESET'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMPMEM_RESET'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMPMEM'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMP_RESET'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_CMP'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_LOCK_WAITS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_LOCKS'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'INNODB_TRX'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'BLACKHOLE'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'ARCHIVE'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'MRG_MYISAM'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'MyISAM'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'MEMORY'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'CSV'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'sha256_password'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'mysql_old_password'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'mysql_native_password'
mysql_1  | 2021-01-05 16:46:46 1 [Note] Shutting down plugin 'binlog'
mysql_1  | 2021-01-05 16:46:46 1 [Note] mysqld: Shutdown complete
mysql_1  |
devilbox_mysql_1 exited with code 1
```
I read this link https://dev.mysql.com/doc/refman/5.6/en/error-creating-innodb.html but didn't get how it could help me.

My main issue is that I have ~20 dbs on 5.6 which I would like to recover and dump, but since I cannot start the image I cannot do it. Is there a way to do it manually in such a situation?
After recovering the dbs I can for sure upgrade everything (docker, devilbox and so on), but I really need to dump all the databases with no loss of data.

Thanks for any help.

**Comment by cytopia** [comment]
I was able to reproduce the behaviour.

### Steps

1. `.env` file: `MYSQL_SERVER=mysql-5.6`
2. docker-compose up -d php httpd bind mysql
3. Create database
4. docker-compose stop
5. `.env` file: `MYSQL_SERVER=mysql-5.7`
6. docker-compose up -d php httpd bind mysql
7. docker-compose stop
8. `.env` file: `MYSQL_SERVER=mysql-5.6`
9. docker-compose up -d php httpd bind mysql

So somehow `docker-compose` is caching the initial volume `devilbox-mysql-5.6` on step `5.` and using it to spin up MySQL 5.7 with it. Then MySQL 5.7 identifies those files as MySQL 5.6 files and auto-upgrades them to MySQL 5.7 files.
If you then start up MySQL 5.6 again, it finds upgraded files and won't start them due to compatibility reasons.

So that being said, you will need to check the documentation on how to downgrade MySQL 5.7 datafiles to MySQL 5.6.
I will investigate why `docker-compose` is caching it, as it should clearly be using a different volume.

---

For future changes (with current behaviour of `docker-compose`) the steps to safely switch databases are as follows: 

1. `.env` file: `MYSQL_SERVER=mysql-5.6`
2. docker-compose up -d php httpd bind mysql
3. Create database
4. docker-compose stop
5. docker-compose rm -f
6. `.env` file: `MYSQL_SERVER=mysql-5.7`
7. docker-compose up -d php httpd bind mysql
8. docker-compose stop
9. docker-compose rm -f
10. `.env` file: `MYSQL_SERVER=mysql-5.6`
11. docker-compose up -d php httpd bind mysql

To sum up the behaviour: It seems you will need to use `docker-compose rm -f` when switching versions.

**Comment by masiorama** [comment]
Hello @cytopia, thanks for the investigation. I'm trying to follow your suggesting about downgrading, but the very first thing that is recommended on mysql docs is to backup the database, which I do not know how to perform since I cannot start mysql5.6, and if I start mysql5.7 the databases are not available... any idea how to proceed avoiding data loss?

**Comment by cytopia** [comment]
Hmm (theoretically), you could `docker-compose rm`, then give it a start with mysql 5.6, stop the container (DO NOT DO a `docker-compose rm` this time) and start up Mysql 5.7. It should then wrongly pick up your old databases.

If that works, you could just enter `mysqldump-secure` and have everything backed up. Also then you can delete your MySQL 5.6 db and start from scratch and import your dump again.

(just in theory though)

**Comment by masiorama** [comment]
Thanks @cytopia, I just tried, and it did partially work... 
Here the logs of the execution of  mysqldump-secure:
```
[INFO]  (SQL):  2/23 Dumping:  db_name (4.21 MB)  Failed
FATAL] (RUN):  2/23 Error dumping db_name
[FATAL] mysqldump: mysqldump  Couldn't execute 'show events'  Cannot proceed because system tables used by Event Scheduler were found damaged at server start (1577)
```
And so on for all the databases.

I noticed that mysql5.7 logs tracked 

```
mysql_1  | 2021-01-07T15:23:30.355740Z 131 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
```
But I do not want to upgrade the databases.
If I check the dump folder I can see the dumps of all the databases, I just tried to load one on an empty db and it worked without errors...

I guess the only thing that is missing now is to clear everything and make mysql5.6 start again  in order to recreate the databases one by one.

**Comment by cytopia** [comment]
Good to hear.

Just to tripple ensure. I would load that dump into a MySQL 5.6 DB or Docker first and try out if it actually works, before wiping ;-)
Once confirmed, you just need to remove the docker volume for MySQL 5.6: (`docker volume ls` and `docker volume rm ...`).
Then you can startup MySQL 5.6 from scratch and import the dump:

https://devilbox.readthedocs.io/en/latest/examples/setup-photon-cms.html

**Comment by cytopia** [comment]
Let me know how that goes. Ticket can be left open as I will update documentation accordingly so that others are aware of this as well

**Comment by masiorama** [comment]
I managed to recover everything. It took me quite some time since I decided to move forward and do many upgrades I had in my list... windows, docker, wsl2 and finally devilbox. Now everything is updated to last version.
I just restored a couple of databases and they work fine with mysql5.6.

Do you think the behavior I encountered was related to my old version of devilbox/docker, or the issue would persist also in the current versions?

Thanks a lot for your help and great work!

**Comment by cytopia** [comment]
Pretty sure this issue will persist if you do not do a `docker-compose rm -f` before switching container version with volumes (mysql, postgres).

**Comment by masiorama** [comment]
@cytopia unfortunately (for me) even using `docker-compose rm -f` before switching container version I ended with all 5.6 databases forced to 5.7 and not usable with 5.6, again... this is indeed a big issue for me since every time I have to backup and to restore a lot of dbs.

**Comment by cytopia** [comment]
This is something I cannot reproduce.

So what I did to test it:

1. I've backed up all data of MySQL and PgSQL
2. I've deleted all docker volumes
    ```bash
    # Remove all volumes (this will destroy all data in persistant volumes and let you start from scratch)
    docker volume ls | grep devilbox | awk '{ print $2 }' | xargs -n1 docker volume rm
    ````
3. Remove stopped docker container
    ```bash
    docker-compose rm -f
    ```
4. Activate MySQL 5.6 in `.env`
    ```bash
    MYSQL_SERVER=mysql-5.6
    ```
5. Start devilbox
    ```bash
    docker-compose up -d php httpd bind mysql
    ````
6. Create a database
7. Stop everything:
    ```bash
    docker-compose stop
    ```
8. Remove stopped docker container
    ```bash
    docker-compose rm -f
    ```
9. Activate MySQL 5.7 in `.env`
    ```bash
    MYSQL_SERVER=mysql-5.7
    ```
10. Start devilbox
    ```bash
    docker-compose up -d php httpd bind mysql
    ````
Now you do not see any previous databases and nothing gets upgraded.
Now when changing databases, you will only have to start from step 3 (deleting the volumes was only to clean everything and start from scratch)

**Comment by masiorama** [comment]
The only difference is that I use `docker-compose down` which, afaik, reading the output, should execute stop and remove.
For the rest, I will try to test again on a different clean devilbox installation asap, thanks.

**Comment by masiorama** [comment]
Actually the problem seems a bit bigger then anticipated.
If I completely remove the images (via docker desktop interface clean up function), and then `docker-compose up mysql`, everything is pulled down with these hashes:
```
devilbox/apache-2.4	IN USE	0.36	b1359ac1d401	about 9 hours ago	170.06 MB	
devilbox/mysql	        IN USE	mysql-5.6-0.7	eeda1a2771c3	12 days ago	302.45 MB	
devilbox/php-fpm   	IN USE	7.3-work-0.122	387b1cdb4fa4	24 days ago	3.26 GB	
cytopia/bind	        IN USE	0.15	ff37cf218d55	over 2 years ago	142.16 MB	
```

Everything starts except for mysql:
```
mysql_1  | 2021-01-12 10:02:22+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.6.50-1debian9 started.
mysql_1  | 2021-01-12 10:02:22+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
mysql_1  | 2021-01-12 10:02:22+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.6.50-1debian9 started.
mysql_1  | 2021-01-12 10:02:22 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
mysql_1  | 2021-01-12 10:02:22 0 [Note] mysqld (mysqld 5.6.50) starting as process 1 ...
mysql_1  | 2021-01-12 10:02:22 1 [Note] Plugin 'FEDERATED' is disabled.
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Using atomics to ref count buffer pool pages
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: The InnoDB memory heap is disabled
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Memory barrier is not used
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Compressed tables use zlib 1.2.11
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Using Linux native AIO
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Using CPU crc32 instructions
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Initializing buffer pool, size = 128.0M
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Completed initialization of buffer pool
mysql_1  | 2021-01-12 10:02:22 1 [Note] InnoDB: Highest supported file format is Barracuda.
mysql_1  | InnoDB: No valid checkpoint found.
mysql_1  | InnoDB: If you are attempting downgrade from MySQL 5.7.9 or later,
mysql_1  | InnoDB: please refer to http://dev.mysql.com/doc/refman/5.6/en/upgrading-downgrading.html
mysql_1  | InnoDB: If this error appears when you are creating an InnoDB database,
mysql_1  | InnoDB: the problem may be that during an earlier attempt you managed
mysql_1  | InnoDB: to create the InnoDB data files, but log file creation failed.
mysql_1  | InnoDB: If that is the case, please refer to
mysql_1  | InnoDB: http://dev.mysql.com/doc/refman/5.6/en/error-creating-innodb.html
mysql_1  | 2021-01-12 10:02:22 1 [ERROR] Plugin 'InnoDB' init function returned error.
mysql_1  | 2021-01-12 10:02:22 1 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
mysql_1  | 2021-01-12 10:02:22 1 [ERROR] Unknown/unsupported storage engine: InnoDB
mysql_1  | 2021-01-12 10:02:22 1 [ERROR] Aborting
mysql_1  |
mysql_1  | 2021-01-12 10:02:22 1 [Note] Binlog end
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'partition'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'PERFORMANCE_SCHEMA'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_DATAFILES'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_TABLESPACES'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN_COLS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_FOREIGN'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_FIELDS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_COLUMNS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_INDEXES'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_TABLESTATS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_SYS_TABLES'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_INDEX_TABLE'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_INDEX_CACHE'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_CONFIG'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_BEING_DELETED'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_DELETED'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_FT_DEFAULT_STOPWORD'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_METRICS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_BUFFER_POOL_STATS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE_LRU'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_BUFFER_PAGE'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX_RESET'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMP_PER_INDEX'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMPMEM_RESET'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMPMEM'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMP_RESET'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_CMP'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_LOCK_WAITS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_LOCKS'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'INNODB_TRX'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'BLACKHOLE'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'ARCHIVE'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'MRG_MYISAM'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'MyISAM'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'MEMORY'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'CSV'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'sha256_password'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'mysql_old_password'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'mysql_native_password'
mysql_1  | 2021-01-12 10:02:22 1 [Note] Shutting down plugin 'binlog'
mysql_1  | 2021-01-12 10:02:22 1 [Note] mysqld: Shutdown complete
mysql_1  |
devilbox_mysql_1 exited with code 1
```

I'm totally lost here. :(

**Comment by cytopia** [comment]
Have you removed the docker volumes (after backing up mysql databases)?

**Comment by masiorama** [comment]
Yes, I did, but just moments ago I found out that I the data purge was hidden in the settings, and there were checkbox for hyper-v and wsl2. I erased everything and I'm starting from scratch again... keep you posted, thanks.

**Comment by masiorama** [comment]
Ok it starts indeed now... I have no evidence that it was not working because of the fact the I was on Hyper-V, then tried to move to wsl2, then reverted to Hyper-V... I am gonna restore one of my dbs, test as you commented above, and give you an update.

**Comment by masiorama** [comment]
I can confirm that it works. and using `stop` and `rm-f` avoid the problem of the unwanted auto upgrade of the data.
Thanks again @cytopia !

**Comment by cytopia** [comment]
Glad to hear. And do use the backup function on a regular base, this way you can always just destroy stuff and start from scratch easily if something breaks.

I will go ahead and document this behaviour so that everyone is aware.

Good troubleshooting!

**Comment by mastrus** [comment]
The solution is to fisically move the file form /var/lib/docker/volumes/devilbox_devilbox-mysql-5.6/_data/myDataBaseNameDirectory to /var/lib/docker/volumes/devilbox_devilbox-mysql-X.X/_data
